tests/MillisecondTimestampBehaviorTest.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace ipl\Tests\Orm;
+
+use DateTime;
+use InvalidArgumentException;
+use ipl\Orm\Behavior\MillisecondTimestamp;
+use PHPUnit\Framework\TestCase;
+
+class MillisecondTimestampBehaviorTest extends TestCase
+{
+    public function testFromDbReturnsNullWhenNullIsPassed()
+    {
+        $this->assertNull((new MillisecondTimestamp([]))->fromDb(null, 'key', null));
+    }
+
+    public function testFromDbReturnsDateTimeInstance()
+    {
+        $behavior = new MillisecondTimestamp([]);
+
+        $this->assertInstanceOf(DateTime::class, $behavior->fromDb(100000, 'key', null));
+        $this->assertInstanceOf(DateTime::class, $behavior->fromDb('100000', 'key', null));
+    }
+
+    public function testToDbThrowsAnExceptionWhenNonDateTimeInstanceIsPassed()
+    {
+        $this->expectException(InvalidArgumentException::class);
+
+        (new MillisecondTimestamp([]))->toDb('test', 'key', null);
+    }
+
+    public function testToDbReturnsReturnsNullWhenNullIsPassed()
+    {
+        $this->assertNull((new MillisecondTimestamp([]))->toDb(null, 'key', null));
+    }
+
+    public function testToDbReturnsTimestampMilliseconds()
+    {
+        $now = new DateTime();
+        $this->assertSame($now->getTimestamp() * 1000.0, (new MillisecondTimestamp([]))->toDb($now, 'key', null));
```

Comment:
Looking at this test, I miss a case for passing an integer or float. Then looking at the implementation, I see that's not supported. Looking again at Icinga DB Web's `Timestamp` behavior, I see support for that, even for strings. Why isn't this included here?

Without this, filtering for this by the user won't be possible.

Reply:
> Looking again at Icinga DB Web's `Timestamp` behavior, I see support for that, even for strings. Why isn't this included here?

We're introducing this so we don't have to mess with strings, floats, etc., so why should we have to support this? Even if `icingadb-web` does this now, we have to adjust everything later so this doesn't happen! I.e. `fromDb()` converts all DB data into a `DateTime` instance, so we can be sure and/or even require that a `DatTime` is passed to `toDb()` then.

Reply:
Well, sure, supporting this in `toDb` is not mandatory. But take a look at the `Binary` behavior included here already. It implements `rewriteCondition` to make filtering work.